### ISSUE TITLE
Can't receive iOS push notifications (development build)

### ISSUE THREAD

**Issue by pliablepixels** [phonegap/phonegap-plugin-push#2078]
### Expected Behaviour

iOS should get push notifications 

### Actual Behaviour

iOS doesn't receive push. Android works just fine. iOS does receive a registration token, though.

### History
My app has been published on the app store for over a year, and I've been using the `1.x` branch of your awesome plugin. I've just migrated to 2.x and firebase messaging.

### Issue research
There are several other issues related to iOS in your issue list but they are either left unresolved due to lack of information, or closed due to user error. I can't see my case mapping to either, but I may be wrong

### Details of problem

I am trying to get iOS and Android push notifications working using Apple's APNs token. From what I've read, this APNs token works irrespective of whether you are in development or production mode. I've generated the APNs key multiple times and uploaded to the firebase console. I've also generated a mobile provisioning profile and installed on my phone.

iOS notifications are never received.

My config.xml for iOS:
```
<platform name="ios">
         <resource-file src="GoogleService-Info.plist" />
```

My Xcode project has Push enabled
<img width="604" alt="screen shot 2017-11-25 at 7 55 38 am" src="https://user-images.githubusercontent.com/4116654/33230687-2635bdd4-d1b6-11e7-8372-810f85d75d5e.png">

I've uploaded the APNs key to firebase (and have validated the App ID prefix matches the app)

<img width="777" alt="screen shot 2017-11-25 at 8 01 46 am" src="https://user-images.githubusercontent.com/4116654/33230745-412a1724-d1b7-11e7-8802-8b506c8b06a6.png">

I've placed `GoogleService-Info.plist` (for iOS) and `google-services.json` (Android) in the root of my project

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
```
iOS 11.1.1
```
#### Cordova CLI version and cordova platform version

    cordova --version                                    7.1.0
    cordova platform version android                    
   android 6.4.0
   ios 4.5.4

#### Plugin version

phonegap-plugin-push 2.1.1 "PushPlugin"

#### Sample Push Data Payload

This is what I'm sending on my server. I've also tried sending test messages from the FCM console. Android always works, iOS never reaches

```
 $json = encode_json ({
        to=>$obj->{token},
        notification=> {
           title=>"For the love of all things sweet, iOS please work",
           body=>"Body"
        },
        data=> {
            title=>$header,
            #body=>"My text",
            icon=>"ic_stat_notification",
            mid=>$mid,

        }
    });
```

As you see I've tried several messages from the console too - never delivers to iOS but says "completed"

<img width="779" alt="screen shot 2017-11-25 at 8 03 20 am" src="https://user-images.githubusercontent.com/4116654/33230751-587df4e0-d1b7-11e7-88d6-f94c531dde00.png">

### Push init code in my app

```
push = PushNotification.init(

                    {
                        "ios":
                        {
                            "alert": true,
                            "badge": true,
                            "sound": ld.soundOnPush,
                            "clearBadge": true,
                           // "fcmSandbox": true // -> my assumption is this is not needed for token mode
                        }
                    }

                );
```

Sample log of token received in iOS: (replaced some characters with stars)

```
2017-11-25 07:33:30.544647-0500 zmNinja[10537:2352990] Nov 25, 2017 07:33 AM DEBUG Push Notification registration ID received: {"registrationId":"cCJ1N******:APA91bEkF-IHMwvPjNvwXgcXeM**********-9NzJvKJZyKJ8rYxtd******-AqJYzGZgWJ******vNugXJL0N7qux-VNYCVw2nrLWygmY8sg0rPGDIJzD9P78-******LFv8g9Mk8KW","registrationType":"FCM"}
```

My push notification handler 
```
push.on('notification', function(data)
{
   console.log (JSON.stringify(data);

}
```

**Comment by pliablepixels** [comment]
Update - I created an archive and uploaded it to test flight. Push notifications work with the production build. Am I wrong in my assumption that the APNs token will work in dev mode.

**Comment by macdonst** [comment]
@pliablepixels you have to make sure if your app is in dev mode that it is signed with a development certificate and you are sending to the development version of APNS. The fact that you are receiving pushes in production mode makes me think it is a mis-match of one of those settings.

**Comment by pliablepixels** [comment]
@macdonst thank you. You said:

> you have to make sure if your app is in dev mode that it is signed with a development certificate and you are sending to the development version of APNS.


1. XCode automatically manages the signing. Here is a screenshot - am I looking at the right place?
<img width="909" alt="screen shot 2017-11-26 at 7 14 38 am" src="https://user-images.githubusercontent.com/4116654/33239876-9bb854a4-d279-11e7-8afd-6df18787f76a.png">

2. On your comment on sending it to the development version of APNS, this part confuses me. I have uploaded the new APNs token to Firebase (not certificates) as I posted earlier. Based on documentation I've read, this token applies to both dev and production. So when you say I have to make sure I am sending it to the development version of APNS, who is "I" ? Is it the mobile app or Firebase? If its the app, how do I make sure (enabling fcmSandbox in init doesn't work)

Thank you.

**Comment by macdonst** [comment]
@pliablepixels 99.98% of the time when folks are having problems receiving the push message it's the certificates. If I were you I'd take FCM out of the scenario for now. Just try and connect to APNS and send a push to the registration ID it should give you a much better idea of what is going on. 

When you are sending to FCM and it forwards along to APNS you may get a false positive. That is they message is successfully received by FCM but makes no guarantee if it gets to the APNS service.

**Comment by pliablepixels** [comment]
gotcha. I'm going to close this for now as unless I get to a point where I can see its a plugin related issue. Thanks for your advice, as always!

**Comment by macdonst** [comment]
@pliablepixels okay, but come back if you don't get it sorted.

**Comment by JaneDawson** [comment]
@pliablepixels: I'm having the exact same issue. Did you find a solution?

**Comment by pliablepixels** [comment]
@JaneDawson - not yet - I've not had a chance to work on this problem since I closed it. If I do, I'll post an update.

**Comment by pliablepixels** [comment]
@JaneDawson - finally got back to it. I'm getting push in both dev and prod mode.
Changes I made: nada/zilch.

Magic. My server message hasn't changed (I went back to putting in the same message I wrote above in this issue). My push handler hasn't changed. My phone gap plugin version hasn't change.

I did however change my phone - from iPhone 7 to X.

At this stage, all I can do is role my eyes :)
